src/nodejs/src/logging.ts

```diff
@@ -4,10 +4,67 @@ import 'logging-init';
 export { LogLevel } from './logging-init';
 export type { LogScope } from './logging-init';
 
+export interface LogRef {
+    target : unknown;
+    id : number;
+}
+
+interface SetItem {
+    ref : LogRef;
+    touchedAt : number;
+}
+
+class LogRefSet {
+    items : SetItem[];
+    capacity : number;
+    idSeed : number;
+
+    constructor(capacity : number) {
+        this.idSeed = 0;
+        this.capacity = capacity;
+        this.items = [];
+    }
+
+    public ref(data: unknown) : LogRef {
```

Comment:
since it's a method in collection class, shouldn't it be name appropriate for collection?